### PR TITLE
Fix error reporting for fetch_key

### PR DESCRIPTION
**Pull request by mscherer**
Since fetch_url already take care of the exception, the try/except
clause is no longer working, so replace it with proper status
checking, thus permitting to remove urlib2 from the import list.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @mscherer. @hacosta please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by hacosta** [comment]
shipit

**Comment by gregdek** [comment]
Thanks again to @mscherer for this PR, and thanks @hacosta for reviewing. Marking for inclusion.

File: packaging/os/rpm_key.py [hunk 1]
```diff
@@ -115,18 +114,18 @@ def __init__(self, module):
...
-            tmpfd, tmpname = tempfile.mkstemp()
-            tmpfile = os.fdopen(tmpfd, "w+b")
-            tmpfile.write(key)
-            tmpfile.close()
-            return tmpname
-        except urllib2.URLError, e:
-            self.module.fail_json(msg=str(e))
+        rsp, info = fetch_url(self.module, url)
+        if info['status'] != 200:
+            self.module.fail_json(msg="failed to fetch key at %s , error was: %s" % (url, info['msg']))
```
Comment (bcoca) [inline packaging/os/rpm_key.py:119]: what if status was redirection?

Comment (mscherer) [inline packaging/os/rpm_key.py:119]: IIRC, urllib2 follow redirections by default. I was unsure on this as well, and did look at others modules to see what they did. For example:

https://github.com/ansible/ansible-modules-core/blob/devel/system/authorized_key.py#L380

https://github.com/ansible/ansible-modules-core/blob/devel/packaging/os/apt_repository.py#L340

And my understanding of the code of fetch_url is that it set the code to be 200 if all is well:

https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/urls.py#L764

So it should be ok (but maybe I am wrong, and that's why we have review)

**Comment by mscherer** [comment]
(as i gave the info, I took the freedom of removing the label on my own PR)

**Comment by gregdek** [comment]
Thanks @mscherer. To the current maintainers, @hacosta please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by hacosta** [comment]
shipit

**Comment by gregdek** [comment]
Thanks again to @mscherer for this PR, and thanks @hacosta for reviewing. Marking for inclusion.

**Pull request merged**